src/Npgsql/SqlQueryParser.cs

```diff
@@ -457,7 +462,20 @@ void MoveToNextStatement()
                     statement = new NpgsqlStatement();
                     statements.Add(statement);
                 }
-                _paramIndexMap.Clear();
+
+                // In positional parameter mode, legacy batching (multi-statement commands) isn't supported, and single-statement commands
+                // don't need to be rewritten.
+                if (positionalParameters)
+                {
+                    if (statementIndex > 0)
```

Comment:
Ok so if there is no option for postional + legacy batching anyway why not key all the important bits off the positional param fact? IMO when a command is found positional we should just pass it all through to pg and let it error out on, if any, things like multiple statements. 

I'm curious to know why you chose to go this route :) Is it a pg protocol - not parser - error in such cases?

Reply:
I thought a lot about this question and went back and forth :) See https://github.com/npgsql/npgsql/pull/3834#issuecomment-861079553.